examples/credel.c

```diff
@@ -0,0 +1,477 @@
+
+    for (i = 0; i < __arraycount(s->group); i++)
+        s->group[i] = H5I_INVALID_HID;
+
+    rc = snprintf(s->output_file, sizeof(s->output_file), "%s.h5", s->progname);
+    if (rc == -1 || rc >= sizeof(s->output_file))
+        errx(EXIT_FAILURE, "output filename was truncated");
+}
+
+static void
+delete_group(state_t *s, const int64_t gidx)
+{
+    hid_t     g;
+    const int ngroups = __arraycount(s->group);
+    char      gname[32];
+
+    assert(0 <= gidx);
+
+    snprintf(gname, sizeof(gname), "/group-%" PRId64, gidx);
+    g = s->group[gidx % ngroups];
+
+    if (H5Ldelete(s->file, gname, H5P_DEFAULT) < 0) {
+        errx(EXIT_FAILURE, "%s: H5Ldelete(, \"%s\", ) failed", __func__, gname);
+    }
+
+    if (H5Gclose(g) < 0)
+        errx(EXIT_FAILURE, "H5Gclose failed");
+
+    if (s->verbose > 0)
+        fprintf(stderr, "Deleted group %s\n", gname);
+
+    s->group[gidx % ngroups] = H5I_INVALID_HID;
+    s->stats.groups.deleted++;
+}
+
+static void
+create_group(state_t *s, const int64_t gidx)
+{
+    const int ngroups = __arraycount(s->group);
+    hid_t     g;
+    char      gname[32];
+
+    assert(0 <= gidx);
+    assert(s->group[gidx % ngroups] == H5I_INVALID_HID);
+
+    snprintf(gname, sizeof(gname), "/group-%" PRId64, gidx);
+    g = H5Gcreate(s->file, gname, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+    if (g < 0)
+        errx(EXIT_FAILURE, "H5Gcreate failed");
+
+    if (s->verbose > 0)
+        fprintf(stderr, "Created group %s\n", gname);
+    s->group[gidx % ngroups] = g;
+    s->stats.groups.created++;
+}
+
+static void
+delete_dataset(state_t *s, const int64_t didx)
+{
+    const int     ndatasets = __arraycount(s->dataset);
+    char          dname[32];
+    const int64_t gidx = didx / 2;
+
+    assert(0 <= gidx && 0 <= didx);
+
+    snprintf(dname, sizeof(dname), "/group-%" PRId64 "/dataset-%" PRId64, gidx, didx);
+    if (H5Ldelete(s->file, dname, H5P_DEFAULT) < 0) {
+        errx(EXIT_FAILURE, "%s: H5Ldelete(, \"%s\", ) failed", __func__, dname);
+    }
+
+    if (s->verbose > 0)
+        fprintf(stderr, "Deleted dataset %s\n", dname);
+#if 1
+    const hid_t ds = s->dataset[didx % ndatasets];
+    if (H5Dclose(ds) < 0)
+        errx(EXIT_FAILURE, "H5Dclose failed");
+#endif
+    s->dataset[didx % ndatasets] = H5I_INVALID_HID;
+    s->stats.datasets.deleted++;
+}
+
+static void
+create_dataset(state_t *s, const int64_t didx)
+{
+    const int     ndatasets = __arraycount(s->dataset);
+    char          dname[32];
+    const int64_t gidx = didx / 2;
+    hid_t         ds;
+
+    assert(0 <= gidx && 0 <= didx);
+    assert(s->dataset[didx % ndatasets] == H5I_INVALID_HID);
+
+    s->dataspace = H5Screate_simple(__arraycount(dims), dims, NULL);
+
+    if (s->dataspace < 0)
+        errx(EXIT_FAILURE, "H5Screate_simple failed");
+
+    snprintf(dname, sizeof(dname), "/group-%" PRId64 "/dataset-%" PRId64, gidx, didx);
+    ds = H5Dcreate2(s->file, dname, H5T_STD_I32BE, s->dataspace, H5P_DEFAULT, s->dcpl, s->dapl);
```

Comment:
Is type conversion an important part of this test?   Why not just use H5T_NATIVE_INT?